PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/sparkserverless/serverexplore/ui/SparkServerlessProvisionDialog.java

```diff
@@ -149,14 +152,31 @@ public void windowOpened(WindowEvent e) {
     }
 
     private void updateAvailableAUAndTotalAU() {
-        ctrlProvider.getAvailableAUAndTotalAU().subscribe(pair -> {
-            availableAUField.setText(String.valueOf(pair.getLeft()));
-            totalAUField.setText(String.valueOf(pair.getRight()));
-        });
+        refreshButton.setEnabled(false);
+        ctrlProvider.getAvailableAUAndTotalAU()
+                .observeOn(ideaSchedulers.dispatchUIThread())
+                .doOnNext(pair -> {
```

Comment:
What's the pair?

Reply:
```
    /**
     * The result of availableAU replies on totalAU, here getAvailableAUAndTotalAU() is defined to make sure that
     * availableAU is calculated after totalAU is ready. Notice that totalAU is calculated once for all,
     * getAvailableAUAndTotalAU() should only be called once, next time getAvailableAU() is enough to calculate availableAU
     */
    public Observable<Pair<Integer, Integer>> getAvailableAUAndTotalAU() {
        return Observable.zip(getTotalAU(), getUsedAU(),
                (totalAU, usedAU) -> Pair.of(getAvailableAU(totalAU, usedAU), totalAU));
    }
```

---
In reply to: [202249077](https://github.com/Microsoft/azure-tools-for-java/pull/1763#discussion_r202249077) [](ancestors = 202249077)

Reply:
The var name should present itself, such as `availableAUAndTotalPair`